tools/Voron.Recovery/RecoveredDatabaseCreator.cs

```diff
@@ -0,0 +1,433 @@
+            }
+            var actions = _databaseDestination.RevisionDocuments();
+            WriteDocumentInternal(document, actions);
+        }
+
+        public void WriteCounterItem(CounterGroupDetail counterGroupDetail, bool skipDocExistance = false)
+        {
+            if (skipDocExistance == false)
+            {
+                using (var tx = _context.OpenWriteTransaction())
+                {
+                    Document originalDoc = _database.DocumentsStorage.Get(_context, counterGroupDetail.DocumentId);
+                    if (originalDoc == null)
+                    {
+                        var orphanDocId = $"{_orphanCountersDocIdPrefix}/{counterGroupDetail.DocumentId}";
+                        var orphanDoc = _database.DocumentsStorage.Get(_context, orphanDocId);
+                        if (orphanDoc == null)
+                        {
+                            using (var doc = _context.ReadObject(
+                                new DynamicJsonValue
+                                {
+                                    ["OriginalDocId"] = counterGroupDetail.DocumentId.ToString(),
+                                    [Constants.Documents.Metadata.Key] = new DynamicJsonValue {[Constants.Documents.Metadata.Collection] = _recoveryLogCollection}
+                                }, orphanDocId, BlittableJsonDocumentBuilder.UsageMode.ToDisk))
+                            {
+                                var _ = _database.DocumentsStorage.Put(_context, orphanDocId, null, doc);
+                            }
+                        }
+
+                        counterGroupDetail.DocumentId.Dispose();
+                        counterGroupDetail.DocumentId = _context.GetLazyString(orphanDocId);
+
+                        tx.Commit();
+                    }
+                }
+            }
+
+            using var actions = _databaseDestination.Counters(_results);
+            actions.WriteCounter(counterGroupDetail);
+        }
+
+        public void WriteAttachment(string hash, string name, string contentType, Stream attachmentStream)
+        {
+            // store this attachment either under relevant orphan doc, or under already seen doc
+            using (var tx = _context.OpenWriteTransaction())
+            {
+                var orphanAttachmentDocId = GetOrphanAttachmentDocId(hash);
+                var orphanAttachmentDoc = _database.DocumentsStorage.Get(_context, orphanAttachmentDocId);
+                if (orphanAttachmentDoc != null)
+                {
+                    // check which documents already seen for this attachment and attach it to them
+                    orphanAttachmentDoc.Data.Modifications = new DynamicJsonValue(orphanAttachmentDoc.Data);
+                    foreach (var docId in orphanAttachmentDoc.Data.GetPropertyNames())
```

Comment:
Let's use a real object here. That would make the code a lot easier and maintainable over time